Server/mods/deathmatch/logic/luadefs/CLuaBlipDefs.cpp

```diff
@@ -70,374 +70,145 @@ void CLuaBlipDefs::AddClass(lua_State* luaVM)
+}
 
-    if (!argStream.HasErrors())
-    {
-        if (CStaticFunctionDefinitions::SetBlipColor(pElement, color))
-        {
-            lua_pushboolean(luaVM, true);
-            return 1;
-        }
-    }
-    else
-        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
+std::uint16_t CLuaBlipDefs::GetBlipVisibleDistance(CBlip* radarMarker) noexcept
+{
+    std::uint16_t visibleDistance;
+    CStaticFunctionDefinitions::GetBlipVisibleDistance(radarMarker, visibleDistance);
 
-    lua_pushboolean(luaVM, false);
-    return 1;
+    return visibleDistance;
 }
 
-int CLuaBlipDefs::SetBlipOrdering(lua_State* luaVM)
+bool CLuaBlipDefs::SetBlipIcon(CElement* radarMarker, std::uint8_t icon)
 {
-    CElement* pElement;
-    int       iOrdering;
+    if (!CBlipManager::IsValidIcon(icon))
+        throw std::invalid_argument("Invalid icon");
 
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadUserData(pElement);
-    argStream.ReadNumber(iOrdering);
+    return CStaticFunctionDefinitions::SetBlipIcon(radarMarker, icon);
+}
 
-    if (!argStream.HasErrors())
+bool CLuaBlipDefs::SetBlipSize(lua_State* luaVM, CElement* radarMarker, std::uint8_t size) noexcept
+{
+    if (size < 0 || size > 25)
     {
-        short sOrdering = Clamp(-32768, iOrdering, 32767);
-
-        if (CStaticFunctionDefinitions::SetBlipOrdering(pElement, sOrdering))
-        {
-            lua_pushboolean(luaVM, true);
-            return 1;
-        }
+        m_pScriptDebugging->LogWarning(luaVM, SString("Blip size beyond 25 is no longer supported (got %i). It will be clamped between 0 and 25.", size));
+        size = Clamp(0ui8, size, 25ui8);
     }
-    else
-        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
 
-    lua_pushboolean(luaVM, false);
-    return 1;
+    return CStaticFunctionDefinitions::SetBlipSize(radarMarker, size);
 }
 
-int CLuaBlipDefs::SetBlipVisibleDistance(lua_State* luaVM)
+bool CLuaBlipDefs::SetBlipColor(CElement* radarMarker, std::uint8_t r, std::uint8_t g, std::uint8_t b, std::uint8_t a) noexcept
 {
-    CElement* pElement;
-    int       iVisibleDistance;
-
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadUserData(pElement);
-    argStream.ReadNumber(iVisibleDistance);
-
-    if (!argStream.HasErrors())
-    {
-        unsigned short usVisibleDistance = Clamp(0, iVisibleDistance, 65535);
+    return CStaticFunctionDefinitions::SetBlipColor(radarMarker, SColorRGBA(r, g, b, a));
+}
 
-        if (CStaticFunctionDefinitions::SetBlipVisibleDistance(pElement, usVisibleDistance))
-        {
-            lua_pushboolean(luaVM, true);
-            return 1;
-        }
-    }
-    else
-        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
+bool CLuaBlipDefs::SetBlipOrdering(CElement* radarMarker, std::int32_t ordering) noexcept
+{
+    return CStaticFunctionDefinitions::SetBlipOrdering(radarMarker, Clamp(0i32, ordering, (std::int32_t)UINT16_MAX));
```

Comment:
`UINT16_MAX` is a macro. You dont need to cast it to `(std::int32_t)`

Reply:
There is `std::numeric_limits<std::int16_t>().max()`